Speech/src/SpeechHelpersTrait.php

```diff
@@ -0,0 +1,115 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Speech;
+
+use Generator;
+use InvalidArgumentException;
+
+/**
+ * Provides helper functions for generated Speech clients.
+ */
+trait SpeechHelpersTrait
+{
+    /**
+     * A list of allowed url schemes.
+     *
+     * @var array
+     */
+    private $urlSchemes = [
+        'gs'
+    ];
+
+    /**
+     * @param string                           $recognitionAudioClass
+     * @param resource|string|RecognitionAudio $audio
+     * @return RecognitionAudio
+     */
+    private function createRecognitionAudioHelper($recognitionAudioClass, $audio)
+    {
+        if (is_object($audio) && $audio instanceof $recognitionAudioClass) {
+            return $audio;
+        }
+        $recognitionAudio = new $recognitionAudioClass();
+        if (is_string($audio)) {
+            if (in_array(parse_url($audio, PHP_URL_SCHEME), $this->urlSchemes)) {
+                $recognitionAudio->setUri($audio);
+            } else {
+                $recognitionAudio->setContent($audio);
+            }
+        } elseif (is_resource($audio)) {
+            $recognitionAudio->setContent(stream_get_contents($audio));
+        } else {
+            throw new InvalidArgumentException(
+                'Given $audio is not valid. ' .
+                'Audio must be a RecognitionAudio ' .
+                'object, a string of bytes, a valid ' .
+                'Google Cloud Storage URI, or a resource.'
+            );
+        }
+        return $recognitionAudio;
+    }
+
+    /**
+     * @param string $requestClass
+     * @param iterable|resource|string $audio
+     * @return StreamingRecognizeRequest[]
+     */
+    private function createStreamingRequestsHelper($requestClass, $audio)
+    {
+        // First, convert string/resource audio into an iterable
+        if (is_string($audio)) {
+            $audio = [$audio];
+        }
+        if (is_resource($audio)) {
+            $audio = $this->createAudioStreamFromResource($audio);
+        }
+
+        // For each chuck in iterable $audio, convert to a request if necessary
+        foreach ($audio as $audioChunk) {
+            if (is_object($audioChunk) && $audioChunk instanceof $requestClass) {
+                yield $audioChunk;
+            } elseif (is_string($audioChunk)) {
+                $request = new $requestClass();
+                $request->setAudioContent($audioChunk);
+                yield $request;
+            } else {
+                throw new InvalidArgumentException(
+                    'Found invalid audio chunk in $audio. ' .
+                    'Audio must be a resource, a string of ' .
+                    'bytes, or an iterable of StreamingRecognizeRequest[] ' .
+                    'or string[].'
+                );
+            }
+        }
+    }
+
+    /**
+     * @param resource $resource
+     * @param int      $chunkSize
```

Comment:
Do you see value in allowing users to configure this?

Reply:
I like that.

Reply:
Fixed